frontera/contrib/backends/remote/messagebus.py

```diff
@@ -41,41 +45,41 @@ def frontier_stop(self):
         self.spider_log_producer.flush()
 
     def add_seeds(self, seeds):
-        per_host = aggregate_per_host(seeds)
-        for host_fprint, host_links in six.iteritems(per_host):
-            self.spider_log_producer.send(host_fprint,
-                                          self._encoder.encode_add_seeds(host_links))
+        key = b'0123456789abcdef0123456789abcdef012345678'
+        message = self._encoder.encode_add_seeds(seeds)
+
+        self.spider_log_producer.send(key, message)
 
     def page_crawled(self, response):
-        host_fprint = get_host_fprint(response)
-        self.spider_log_producer.send(host_fprint, self._encoder.encode_page_crawled(response))
+        key = get_host_fprint(response)
+        message = self._encoder.encode_page_crawled(response)
+
+        self.spider_log_producer.send(key, message)
 
     def links_extracted(self, request, links):
-        per_host = aggregate_per_host(links)
-        for host_fprint, host_links in six.iteritems(per_host):
-            self.spider_log_producer.send(host_fprint,
-                                          self._encoder.encode_links_extracted(request, host_links))
+        key = get_host_fprint(request)
+        message = self._encoder.encode_links_extracted(request, links)
+
+        self.spider_log_producer.send(key, message)
 
     def request_error(self, page, error):
-        host_fprint = get_host_fprint(page)
-        self.spider_log_producer.send(host_fprint, self._encoder.encode_request_error(page, error))
+        key = get_host_fprint(page)
+        message = self._encoder.encode_request_error(page, error)
 
-    def _get_next_requests(self, max_n_requests, **kwargs):
-        requests = []
+        self.spider_log_producer.send(key, message)
+
+    def get_next_requests(self, max_n_requests, **kwargs):
         for encoded in self.consumer.get_messages(count=max_n_requests, timeout=self._get_timeout):
             try:
-                request = self._decoder.decode_request(encoded)
-            except Exception as exc:
-                self._logger.warning("Could not decode message: {0}, error {1}".format(encoded, str(exc)))
-            else:
-                requests.append(request)
-        self.spider_log_producer.send(b'0123456789abcdef0123456789abcdef012345678',
-                                      self._encoder.encode_offset(self.partition_id,
-                                                                  self.consumer.get_offset(self.partition_id)))
-        return requests
+                yield self._decoder.decode_request(encoded)
```

Comment:
Got the point, will check this. I believe, I've seen somewhere that get_next_requests should be iterable, not just list. From my point of view, making it iterable might open path to different kind of asynchronous backends in the future.